spacksites/spack-env-templates/slurm/avx2.yaml

```diff
@@ -0,0 +1,45 @@
+spack:
+  concretizer:
+    unify: when_possible
+  view: false
+
+  packages:
+    all:
+      target: [x86_64_v3]
+
+  # the Michael avx2 end applications and Young GPU applications
+  specs:
+  - matrix:
+    - - bcftools@1.16
+      - beast2@2.7.4
+      - bedtools2@2.31.0
+      - bwa@0.7.17
+      - castep@24.1 +libxc ^openblas threads=none
+      - castep@23.1 +libxc ^openblas threads=none
+      - cp2k@2024.3
+      - gatk@4.4.0.0
+      - gromacs@2024.3 -double
+      - gromacs@2023.5 -double +plumed
+      - gsl@2.8
+      - hdf5@1.14.3 +cxx +fortran -mpi +hl
+      - hdf5@1.14.3 +fortran +mpi +hl
+      - htslib@1.17
+      - lammps@20240829.1 +mpi +python +amoeba +asphere +bocs +body +bpm +brownian +cg-dna +cg-spica +class2 +colloid +colvars +compress +coreshell +dielectric +diffraction +dipole +dpd-basic +dpd-meso +dpd-react +dpd-smooth +drude +eff +electrode +extra-compute +extra-dump +extra-fix +extra-molecule +extra-pair +fep +granular +interlayer +kspace +lepton +machdyn +manybody +mc +meam +mesont +misc +ml-iap +ml-pod +ml-snap +ml-uf3 +mofff +molecule +openmp +opt +orient +peri +phonon +plugin +poems +qeq +reaction +reaxff +replica +rigid +rheo +shock +sph +spin +srd +tally +uef +voronoi +yaff
+      - libxc@6.2.2
+      - namd@2.14 ^charmpp backend=ucx pmi=pmix ^fftw -mpi
+      - namd@3.0.1 ^charmpp backend=ucx pmi=pmix ^fftw -mpi
+      - namd@3.0.1 ^charmpp backend=verbs ^fftw -mpi
+      - netcdf-c@4.9.2 -mpi
+      - netcdf-fortran@4.6.1
+      - netlib-scalapack@2.2.0
+      - numactl@2.0.18
+      - orca@avx2-6.0.1
+      - picard@2.26.2
+      - plumed@2.9.2
+      - samtools@1.17
+    - - '%gcc@12.3.0'
+
+  include_concrete:
+  - ${SPACK_ROOT}/var/spack/environments/gpu_first
+  - ${SPACK_ROOT}/var/spack/environments/gpu-on-gpu
+  - ${SPACK_ROOT}/var/spack/environments/gpu_third
```

Comment:
I don't think using ${SPACK_ROOT} here works, which was why I had to go to full paths for the other environments.

Need to make sure that these get set wherever castep, orca and vasp packages get installed (I guess could go into the various `settings/initial_site_packages_` yamls now):

```
  packages:
    castep:
      permissions:
        read: group
        group: ag-archpc-castep
    orca:
      permissions:
        read: group
        group: ag-archpc-orca
    vasp@6:
      permissions:
        read: group
        group: ag-archpc-vasp6
```

Reply:
> I don't think using ${SPACK_ROOT} here works, which was why I had to go to full paths for the other environments.

Agreed, I [mentioned](https://github.com/UCL-ARC/hpc-spack/pull/68#discussion_r2104301091) this was mainly a placeholder 🙂 But it's still not clear to me what should be the absolute paths.

> Need to make sure that these get set wherever castep, orca and vasp packages get installed 

Done!